### PR TITLE
[Mobile] Copy/Paste improvement

### PR DESCRIPTION
**Pull request by geriux**
## Description
This PR adds a `Clipboard` context instead of using the `core/editor` store, I ended up not using the actual clipboard of the device for now because the current implementation checks the clipboard when trying to add a new block or opening the toolbar options, this on iOS wouldn't be a good approach since you'll see a prompt notice about using the device clipboard:

<kbd></kbd>

It also changes how the `Notice` component stores its data, it was using the `core/editor` store but now it'll use `core/notices`.

These changes are in favor of the [issue with dependencies](https://github.com/WordPress/gutenberg/pull/26954#issuecomment-726678393) in the `Inserter` and `BlockMobileToolbar`.

## How has this been tested?

<details><summary>Steps to test</summary>
 
**Copy / Paste after block**

- Open the demo app with metro running
- Select one of the blocks
- Tap on the three dots button (bottom right of the block)
- Tap on **Copy**
- Tap on the three dots button again (bottom right of the block)
- Tap on **Paste after**
- **Expect** to see your copied block pasted after the previously selected block

**Cut block**

- Open the demo app with metro running
- Select one of the blocks
- Tap on the three dots button (bottom right of the block)
- Tap on **Cut**
- **Expect** to see the block disappear
- Add a Group block
- Tap on the + button to bring up the inserter
- **Expect** to see **Copied block** in the first item of the menu
- Tap on **Copied block**
- **Expect** to see your copied block pasted into the Group block

**Duplicate block**

- Open the demo app with metro running
- Select one of the blocks
- Tap on the three dots button (bottom right of the block)
- Tap on **Duplicate**
- **Expect** to see your block duplicated

For the above actions make sure:

- [ ] Notice is showing 
- [ ] Self dismissing within 3 seconds
- [ ] Dismissing immediately when tapped
- [ ] Width adjusts for landscape/portrait

</details>

## Screenshots 
N/A

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/mobile/clipboard/index.native.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ */
+import { createContext, useCallback, useState } from '@wordpress/element';
+
+export const ClipboardContext = createContext( {} );
+const { Provider, Consumer } = ClipboardContext;
+export { Consumer as ClipboardConsumer };
+
+export default function ClipboardProvider( { children } ) {
+	const [ clipboard, setClipboard ] = useState();
+	const updateClipboard = useCallback( ( clipboardUpdate ) => {
```
Comment (Tug) [inline packages/components/src/mobile/clipboard/index.native.js:12]: Isn't `setClipboard` going to be the same reference on each call? In that case we don't need that `updateClipboard` method I think

File: packages/block-editor/src/components/inserter/menu.native.js [hunk 2]
```diff
@@ -125,48 +156,55 @@ export class InserterMenu extends Component {
...
-								numColumns={ numberOfColumns }
-								data={ items }
-								ItemSeparatorComponent={ () => (
-									<TouchableWithoutFeedback
-										accessible={ false }
-									>
-										<View style={ styles.rowSeparator } />
-									</TouchableWithoutFeedback>
+			<ClipboardConsumer>
+				{ ( { clipboard } ) => (
```
Comment (Tug) [inline packages/block-editor/src/components/inserter/menu.native.js:163]: I wonder if this is the best approach as it's quite some boilerplate for just retrieving the clipboard. Also we don't really need to "react" on clipboard change here, the clipboard state will always be modified before opening the inserter.
Maybe a simple singleton object (like `isRTL` in `@wordpress/i18n`) would have been simpler to store clipboard data and allowed us to get the clipboard content outside a component's render method.
Any advantage I'm missing in using the Provider/Consumer approach?

**Review by Tug — COMMENTED** [review]
Tested and this works well 🎉 
I had a few questions about your approach.

File: packages/block-editor/src/components/inserter/menu.native.js (hunk 2 above)
Comment (geriux) [reply packages/block-editor/src/components/inserter/menu.native.js]: Thank you for your suggestion Tug! I updated the code to use a singleton approach instead.

File: packages/components/src/mobile/clipboard/index.native.js (hunk 1 above)
Comment (geriux) [reply packages/components/src/mobile/clipboard/index.native.js]: I updated this code, removing the provider/context approach

File: packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js [hunk 3]
```diff
@@ -115,16 +121,37 @@ const BlockActionsMenu = ( {
...
 
+	function onPasteBlock() {
+		if ( ! clipboard ) {
+			return;
+		}
+
+		pasteBlock( rawHandler( { HTML: clipboard } )[ 0 ] );
+	}
+
+	function isPasteEnabled() {
```
Comment (Tug) [inline packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js:137]: It doesn't look like this is used anywhere else. Could we make it a `const`?

File: packages/block-editor/src/components/inserter/menu.native.js [hunk 4]
```diff
@@ -115,6 +116,37 @@ export class InserterMenu extends Component {
 		this.setState( { numberOfColumns, itemWidth, maxWidth } );
 	}
 
+	getItems() {
```
Comment (Tug) [inline packages/block-editor/src/components/inserter/menu.native.js:119]: Could we add a comment here explaining why we need to pre-process the items list? (to add the paste option)

File: packages/editor/src/store/actions.native.js [hunk 5]
```diff
@@ -44,43 +39,3 @@ export function updateClipboard( clipboard ) {
 		clipboard,
```
Comment (Tug) [inline packages/editor/src/store/actions.native.js:39]: Shouldn't we remove `updateClipboard` as well here?

File: packages/editor/src/store/selectors.native.js [hunk 6]
```diff
@@ -66,14 +66,3 @@ export const isEditedPostAutosaveable = createRegistrySelector(
 export function getClipboard( state ) {
```
Comment (Tug) [inline packages/editor/src/store/selectors.native.js:66]: Same for `getClipboard` ?

**Review by Tug — COMMENTED** [review]
Code looks good, I had a few minor comments

File: packages/block-editor/src/components/inserter/menu.native.js (hunk 4 above)
Comment (geriux) [reply packages/block-editor/src/components/inserter/menu.native.js:124]: Of course, I added a comment 👍

File: packages/editor/src/store/actions.native.js (hunk 5 above)
Comment (geriux) [reply packages/editor/src/store/actions.native.js]: Right! Good catch

File: packages/editor/src/store/selectors.native.js (hunk 6 above)
Comment (geriux) [reply packages/editor/src/store/selectors.native.js]: I removed this one as well

File: packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js (hunk 3 above)
Comment (geriux) [reply packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js:137]: What do you mean by making it a `const`?

Comment (Tug) [reply packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js:137]: I mean it's only called once in the `options` declaration a few lines above, we don't need to make it a function.
```js
const clipboardBlock = rawHandler( { HTML: clipboard } )[ 0 ];
const isPasteEnabled = clipboardBlock && canInsertBlockType( clipboardBlock.name, rootClientId )
```

Comment (geriux) [reply packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js]: Gotcha, thanks! Updated it.

One thing though, I had to add a local state to the Block Actions menu component to store the current clipboard. For cases like:

Select a block > Open the actions menu > Tap on copy > Open the actions menu of the same block > `Paste after` action should show up.

Comment (Tug) [reply packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js]: It's a bit confusing but I guess it's a cool workaround :+1:
Let's revisit when we actually use the phone clipboard

**Review by Tug — APPROVED** [review]

**Pull request merged**